packages/ilios-common/addon/components/offering-form.js

```diff
@@ -149,19 +150,77 @@ export default class OfferingForm extends Component {
     return DEFAULT_URL_VALUE;
   }
 
-  load = restartableTask(async (element, [offering, cohorts]) => {
-    await this.loadData.perform(offering, cohorts);
-    await timeout(1);
-  });
+  @cached
+  get offeringFormData() {
+    return new TrackedAsyncData(this.loadData(this.args.offering, this.args.cohorts));
+  }
 
-  loadData = restartableTask(async (offering, cohorts) => {
+  async loadData(offering, cohorts) {
     this.availableInstructorGroups = await this.loadAvailableInstructorGroups(cohorts);
 
     if (isPresent(offering)) {
       await this.loadAttrsFromOffering.perform(offering);
     } else {
       this.loadDefaultAttrs();
     }
+  }
+
+  async loadAvailableInstructorGroups(cohorts) {
+    let associatedSchools;
+    if (isEmpty(cohorts)) {
+      associatedSchools = [];
+    } else {
+      const cohortSchools = await map(cohorts, async (cohort) => {
+        const programYear = await cohort.programYear;
+        const program = await programYear.program;
+        return program.school;
+      });
+      associatedSchools = uniqueValues(cohortSchools);
+    }
+    const allInstructorGroups = await map(associatedSchools, (school) => school.instructorGroups);
+    return allInstructorGroups.reduce((flattened, arr) => {
+      return [...flattened, ...arr];
+    }, []);
+  }
+
+  loadDefaultAttrs() {
+    if (this.loaded) {
```

Comment:
I think this `loaded` attribute can be removed entirely - it looks like we're tracking state for no good reason with this.